ccl/logictestccl: TestCCLLogic failed

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3257369&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3257369&tab=artifacts#/) on master @ [57ec5f24dc81f947231b877969b7b1603e8721b4](https://github.com/cockroachdb/cockroach/commits/57ec5f24dc81f947231b877969b7b1603e8721b4):


```
=== RUN   TestCCLLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic370260296
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3353: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic370260296
--- FAIL: TestCCLLogic (179.23s)
=== RUN   TestCCLLogic/fakedist-metadata/partitioning_enum/drop_enum_partitioning_value
E210803 19:33:03.315686 96402 jobs/adopt.go:327  [n1] 62  job 681406572588105729: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@primary
E210803 19:33:03.701026 97478 jobs/adopt.go:327  [n1] 63  job 681406574199996417: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@primary
E210803 19:33:04.217952 93421 jobs/adopt.go:327  [n1] 64  job 681406575863136257: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@idx
E210803 19:33:04.606138 95974 sql/logictest/logic.go:3561  [-] 65  
E210803 19:33:04.606138 95974 sql/logictest/logic.go:3561  [-] 65 +
E210803 19:33:04.606138 95974 sql/logictest/logic.go:3561  [-] 65 +testdata/logic_test/partitioning_enum:95: 
E210803 19:33:04.606138 95974 sql/logictest/logic.go:3561  [-] 65 +expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
    logic.go:1818: 
         
        testdata/logic_test/partitioning_enum:95: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: testdata/logic_test/partitioning_enum with config fakedist-metadata: 15 tests, 1 failures
--- total progress: 545 statements/queries
            --- FAIL: TestCCLLogic/fakedist-metadata/partitioning_enum/drop_enum_partitioning_value (1.65s)
=== RUN   TestCCLLogic/fakedist-metadata/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-metadata/partitioning_enum
=== CONT  TestCCLLogic/fakedist-metadata/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-metadata/partitioning_enum (4.04s)
=== RUN   TestCCLLogic/fakedist-metadata
    --- FAIL: TestCCLLogic/fakedist-metadata (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3320252&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3320252&tab=artifacts#/) on master @ [d13d9cfd5b379b0636f7146d8c1bbbbcf0709d08](https://github.com/cockroachdb/cockroach/commits/d13d9cfd5b379b0636f7146d8c1bbbbcf0709d08):


```
=== RUN   TestCCLLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic576804854
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3337: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic576804854
--- FAIL: TestCCLLogic (247.72s)
=== RUN   TestCCLLogic/fakedist-spec-planning/partitioning_enum/drop_enum_partitioning_value
E210818 15:39:06.204970 124589 jobs/adopt.go:424  [n1] 33  job 685607308495192065: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@primary
E210818 15:39:06.755121 125787 jobs/adopt.go:424  [n1] 34  job 685607310660075521: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@primary
E210818 15:39:07.420450 126283 jobs/adopt.go:424  [n1] 35  job 685607312551378945: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@idx
E210818 15:39:08.086499 124338 sql/logictest/logic.go:3545  [-] 36  
E210818 15:39:08.086499 124338 sql/logictest/logic.go:3545  [-] 36 +
E210818 15:39:08.086499 124338 sql/logictest/logic.go:3545  [-] 36 +testdata/logic_test/partitioning_enum:95: 
E210818 15:39:08.086499 124338 sql/logictest/logic.go:3545  [-] 36 +expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
    logic.go:1813: 
         
        testdata/logic_test/partitioning_enum:95: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: testdata/logic_test/partitioning_enum with config fakedist-spec-planning: 15 tests, 1 failures
--- total progress: 589 statements/queries
            --- FAIL: TestCCLLogic/fakedist-spec-planning/partitioning_enum/drop_enum_partitioning_value (2.22s)
=== RUN   TestCCLLogic/fakedist-spec-planning/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-spec-planning/partitioning_enum
=== CONT  TestCCLLogic/fakedist-spec-planning/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-spec-planning/partitioning_enum (5.54s)
=== RUN   TestCCLLogic/fakedist-spec-planning
    --- FAIL: TestCCLLogic/fakedist-spec-planning (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3321593&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3321593&tab=artifacts#/) on master @ [9d76ce56296be716dc2e1026ab99f8fcd0c19ab7](https://github.com/cockroachdb/cockroach/commits/9d76ce56296be716dc2e1026ab99f8fcd0c19ab7):


```
=== RUN   TestCCLLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic860085847
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3337: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic860085847
--- FAIL: TestCCLLogic (244.78s)
=== RUN   TestCCLLogic/fakedist-metadata/partitioning_enum/drop_enum_partitioning_value
E210818 19:47:48.585461 101073 jobs/adopt.go:424  [n1] 25  job 685656206542831617: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@primary
E210818 19:47:49.360337 102456 jobs/adopt.go:424  [n1] 26  job 685656208973365249: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@primary
E210818 19:47:50.323998 103239 jobs/adopt.go:424  [n1] 27  job 685656212002373633: adoption completed with error could not remove enum value "a" as it is being used in the partitioning of index tbl@idx
E210818 19:47:50.859461 100749 sql/logictest/logic.go:3545  [-] 28  
E210818 19:47:50.859461 100749 sql/logictest/logic.go:3545  [-] 28 +
E210818 19:47:50.859461 100749 sql/logictest/logic.go:3545  [-] 28 +testdata/logic_test/partitioning_enum:95: 
E210818 19:47:50.859461 100749 sql/logictest/logic.go:3545  [-] 28 +expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
    logic.go:1813: 
         
        testdata/logic_test/partitioning_enum:95: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: testdata/logic_test/partitioning_enum with config fakedist-metadata: 15 tests, 1 failures
--- total progress: 545 statements/queries
            --- FAIL: TestCCLLogic/fakedist-metadata/partitioning_enum/drop_enum_partitioning_value (2.50s)
=== RUN   TestCCLLogic/fakedist-metadata/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-metadata/partitioning_enum
=== CONT  TestCCLLogic/fakedist-metadata/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-metadata/partitioning_enum (5.73s)
=== RUN   TestCCLLogic/fakedist-metadata
    --- FAIL: TestCCLLogic/fakedist-metadata (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3322557&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3322557&tab=artifacts#/) on master @ [a72c83fcfab599c45ee3eddc73bb1175c136889b](https://github.com/cockroachdb/cockroach/commits/a72c83fcfab599c45ee3eddc73bb1175c136889b):


```
=== RUN   TestCCLLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic536489201
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3337: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic536489201
I210818 22:27:42.556170 294130 2@util/log/event_log.go:32  [n7] 2 ={"Timestamp":1629325662556167087,"EventType":"runtime_stats","MemRSSBytes":6843973632,"GoroutineCount":3778,"MemStackSysBytes":88014848,"GoAllocBytes":650670512,"GoTotalBytes":5803033448,"GoStatsStaleness":0.016893804,"HeapFragmentBytes":409112144,"HeapReservedBytes":4370857984,"HeapReleasedBytes":856686592,"CGoAllocBytes":88999456,"CGoTotalBytes":131702784,"CGoCallRate":118.72917,"CPUUserPercent":163.41434,"CPUSysPercent":13.037319,"GCPausePercent":0.097893424,"GCRunCount":123,"NetHostRecvBytes":21115604,"NetHostSendBytes":21104276}
--- FAIL: TestCCLLogic (351.11s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (0.00s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_placement_restricted
=== PAUSE TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_placement_restricted
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_placement_restricted
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_placement_restricted
    logic.go:1491: 
        :0: error while processing
    logic.go:1491: pq: setting updated but timed out waiting to read new value
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_placement_restricted (46.28s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3322897&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3322897&tab=artifacts#/) on master @ [df6603d0093fbb7f801f287b3126f11dabc6b9d5](https://github.com/cockroachdb/cockroach/commits/df6603d0093fbb7f801f287b3126f11dabc6b9d5):


```
--- progress: testdata/logic_test/multi_region: 31 statements/queries
--- progress: testdata/logic_test/multi_region_zone_configs: 47 statements/queries
--- total progress: 791 statements/queries
E210819 00:35:21.210260 409563 jobs/registry.go:771  [-] 102  error expiring job sessions: expire-sessions: context canceled
E210819 00:35:21.210374 409563 jobs/registry.go:779  [-] 103  failed to serve pause and cancel requests: context canceled
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104  error executing schedules: scheduler-stats: failed to send RPC: sending to all replicas failed; last error: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:37702->127.0.0.1:35569: use of closed network connection
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +(1) attached stack trace
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  -- stack trace:
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:673
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:274
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:325
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:431
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowExWithCols
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:483
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryRowEx
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:469
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/jobs.newLoopStats
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/job_scheduler.go:208
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/jobs.(*jobScheduler).executeSchedules
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/job_scheduler.go:267
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/jobs.(*jobScheduler).runDaemon.func1.1
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/job_scheduler.go:368
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:839
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:879
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:838
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/jobs.(*jobScheduler).runDaemon.func1
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/job_scheduler.go:367
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | runtime.goexit
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +Wraps: (2) scheduler-stats
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +Wraps: (3) failed to send RPC: sending to all replicas failed; last error: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:37702->127.0.0.1:35569: use of closed network connection
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  |
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | (opaque error leaf)
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +  | type name: github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/kvcoord.sendError
E210819 00:35:21.213826 413750 jobs/job_scheduler.go:371  [n8] 104 +Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueLeaf
--- progress: testdata/logic_test/multi_region_backup: 31 statements/queries
E210819 00:35:22.399693 227054 jobs/registry.go:805  [-] 105  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
--- progress: testdata/logic_test/multi_region_privileges: 11 statements/queries
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/multi_region_import_export (76.22s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3352057&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3352057&tab=artifacts#/) on master @ [967ed00f80981ce8848a5e8144ee6fbd29bc95bb](https://github.com/cockroachdb/cockroach/commits/967ed00f80981ce8848a5e8144ee6fbd29bc95bb):


```
--- progress: testdata/logic_test/multi_region: 146 statements/queries
--- progress: testdata/logic_test/regional_by_row: 75 statements/queries
--- progress: testdata/logic_test/multi_region_backup: 82 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 73 statements/queries
--- progress: testdata/logic_test/multi_region: 156 statements/queries
--- progress: testdata/logic_test/multi_region_backup: 94 statements/queries
--- progress: testdata/logic_test/multi_region_zone_configs: 95 statements/queries
E210825 04:21:30.223408 1380677 jobs/adopt.go:410  [n1] 81  job 687455895180902401: adoption completed with error could not remove enum value "ap-southeast-2" as it is the home region for table "t"
--- progress: testdata/logic_test/regional_by_row: 115 statements/queries
--- progress: testdata/logic_test/multi_region: 169 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 75 statements/queries
--- progress: testdata/logic_test/multi_region_zone_configs: 103 statements/queries
E210825 04:21:32.403709 1260780 jobs/adopt.go:410  [n1] 82  job 687455866546847745: adoption completed with error attempting to GC indexes: removing index 2 zone configs: relation "[58]" does not exist
E210825 04:21:32.580929 1260760 jobs/adopt.go:410  [n1] 83  job 687455866435796993: adoption completed with error attempting to GC indexes: removing index 1 zone configs: relation "[58]" does not exist
--- progress: testdata/logic_test/multi_region_backup: 107 statements/queries
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_placement_restricted (79.00s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/multi_region_privileges
=== PAUSE TestCCLLogic/multiregion-9node-3region-3azs/multi_region_privileges
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/multi_region_privileges
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/multi_region_privileges
    logic.go:1276: 
        testdata/logic_test/multi_region_privileges:12: error while processing
    logic.go:1276: pq: internal error while retrieving user account: operation "get-user-timeout" timed out after 10s: context deadline exceeded
--- done: testdata/logic_test/multi_region_privileges with config multiregion-9node-3region-3azs: 1 tests, 0 failures
E210825 04:20:34.637921 415480 jobs/registry.go:779  [-] 42  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E210825 04:20:34.649474 212410 jobs/registry.go:779  [-] 43  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E210825 04:20:34.718972 259729 jobs/registry.go:771  [-] 44  error expiring job sessions: expire-sessions: context canceled
E210825 04:20:34.777888 259729 jobs/registry.go:779  [-] 45  failed to serve pause and cancel requests: context canceled
E210825 04:20:34.817932 224110 jobs/registry.go:771  [-] 46  error expiring job sessions: expire-sessions: context canceled
E210825 04:20:34.818319 224110 jobs/registry.go:779  [-] 47  failed to serve pause and cancel requests: context canceled
E210825 04:20:34.836942 415482 jobs/registry.go:792  [-] 48  error claiming jobs: could not query jobs table: claim-jobs: context canceled
E210825 04:20:34.921316 317777 jobs/registry.go:805  [-] 49  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: context canceled
E210825 04:20:34.986141 282605 jobs/registry.go:771  [-] 50  error expiring job sessions: expire-sessions: context canceled
E210825 04:20:34.987551 282605 jobs/registry.go:779  [-] 51  failed to serve pause and cancel requests: context canceled
--- progress: testdata/logic_test/multi_region_import_export: 11 statements/queries
E210825 04:20:40.008134 776575 kv/kvserver/queue.go:1098  [n1,raftlog,s1,r4/1:/System{/tsd-tse}] 52  failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
E210825 04:20:40.011499 186239 kv/kvserver/reports/reporter.go:143  [n1,replication-reporter] 53  failed to generate replication reports: failed to compute constraint conformance report: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
--- progress: testdata/logic_test/multi_region_backup: 6 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 26 statements/queries
--- progress: testdata/logic_test/regional_by_row_rename_column: 6 statements/queries
--- progress: testdata/logic_test/multi_region_default_primary_region: 6 statements/queries
--- progress: testdata/logic_test/multi_region_zone_configs: 3 statements/queries
--- progress: testdata/logic_test/multi_region_default_primary_region: 7 statements/queries
--- progress: testdata/logic_test/multi_region_tpcc: 1 statements/queries
--- progress: testdata/logic_test/multi_region_tpcc: 2 statements/queries
--- progress: testdata/logic_test/global_placement_restricted: 1 statements/queries
--- progress: testdata/logic_test/multi_region_default_primary_region: 11 statements/queries
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/multi_region_privileges (80.17s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3400862&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3400862&tab=artifacts#/) on master @ [e931244b59d349ddc5ee05ae319f9b6768601e0b](https://github.com/cockroachdb/cockroach/commits/e931244b59d349ddc5ee05ae319f9b6768601e0b):


```
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3388: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic755978830
--- FAIL: TestCCLLogic (508.43s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (215.41s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_query_behavior
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46  
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +testdata/logic_test/regional_by_row_query_behavior:463: SELECT message FROM [SHOW KV TRACE FOR SESSION] WITH ORDINALITY
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 + WHERE message LIKE 'fetched:%' OR message LIKE 'output row%'
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 + OR message LIKE 'Scan%'
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 + ORDER BY ordinality ASC
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +expected:
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    Scan /Table/56/1/"@"/10/0
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    Scan /Table/56/1/"\xc0"/10/0
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    Scan /Table/56/1/"\x80"/10/0
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    output row: [10 10 11 12 NULL]
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +but found (query options: "rowsort" -> ignore the following ordering of rows) :
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    Scan /Table/56/1/"@"/10/0
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    Scan /Table/56/1/"\x80"/10/0
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
E210902 20:28:05.361756 448893 sql/logictest/logic.go:3596  [-] 46 +    output row: [10 10 11 12 NULL]
    logic.go:2311: 
         
        testdata/logic_test/regional_by_row_query_behavior:463: SELECT message FROM [SHOW KV TRACE FOR SESSION] WITH ORDINALITY
         WHERE message LIKE 'fetched:%' OR message LIKE 'output row%'
         OR message LIKE 'Scan%'
         ORDER BY ordinality ASC
        expected:
            Scan /Table/56/1/"@"/10/0
            Scan /Table/56/1/"\xc0"/10/0
            Scan /Table/56/1/"\x80"/10/0
            fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
            output row: [10 10 11 12 NULL]
            
        but found (query options: "rowsort" -> ignore the following ordering of rows) :
            Scan /Table/56/1/"@"/10/0
            Scan /Table/56/1/"\x80"/10/0
            fetched: /regional_by_row_table/primary/'ca-central-1'/10/pk2/a/b -> /10/11/12
            output row: [10 10 11 12 NULL]
--- done: testdata/logic_test/regional_by_row_query_behavior with config multiregion-9node-3region-3azs: 34 tests, 1 failures
E210902 20:28:05.392363 448893 sql/logictest/logic.go:3621  [-] 47  testdata/logic_test/regional_by_row_query_behavior:475: too many errors encountered, skipping the rest of the input
    logic.go:3156: 
        testdata/logic_test/regional_by_row_query_behavior:475: error while processing
    logic.go:3156: testdata/logic_test/regional_by_row_query_behavior:475: too many errors encountered, skipping the rest of the input
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_query_behavior (4.50s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472412&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3472412&tab=artifacts#/) on master @ [af8d186483d460e244073fe49fade1d7d1467d45](https://github.com/cockroachdb/cockroach/commits/af8d186483d460e244073fe49fade1d7d1467d45):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic584290739
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3436: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic584290739
--- FAIL: TestCCLLogic (481.05s)
=== RUN   TestCCLLogic/fakedist-disk/partitioning_enum/drop_enum_partitioning_value
--- total progress: 570 statements/queries
    logic.go:1867: 
         
        testdata/logic_test/partitioning_enum:95: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: testdata/logic_test/partitioning_enum with config fakedist-disk: 15 tests, 1 failures
            --- FAIL: TestCCLLogic/fakedist-disk/partitioning_enum/drop_enum_partitioning_value (1.55s)
=== RUN   TestCCLLogic/fakedist-disk/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-disk/partitioning_enum
=== CONT  TestCCLLogic/fakedist-disk/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-disk/partitioning_enum (4.10s)
=== RUN   TestCCLLogic/fakedist-disk
    --- FAIL: TestCCLLogic/fakedist-disk (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3482602&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3482602&tab=artifacts#/) on master @ [ecf0ed843be08170dc825467936d3dcb3cb05c62](https://github.com/cockroachdb/cockroach/commits/ecf0ed843be08170dc825467936d3dcb3cb05c62):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic278718597
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3436: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic278718597
--- FAIL: TestCCLLogic (550.65s)
=== RUN   TestCCLLogic/local-vec-off/partitioning_enum/drop_enum_partitioning_value
    logic.go:1867: 
         
        testdata/logic_test/partitioning_enum:92: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: testdata/logic_test/partitioning_enum with config local-vec-off: 15 tests, 1 failures
--- total progress: 437 statements/queries
            --- FAIL: TestCCLLogic/local-vec-off/partitioning_enum/drop_enum_partitioning_value (1.51s)
=== RUN   TestCCLLogic/local-vec-off/partitioning_enum
=== PAUSE TestCCLLogic/local-vec-off/partitioning_enum
=== CONT  TestCCLLogic/local-vec-off/partitioning_enum
        --- FAIL: TestCCLLogic/local-vec-off/partitioning_enum (3.87s)
=== RUN   TestCCLLogic/local-vec-off
    --- FAIL: TestCCLLogic/local-vec-off (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries mberhault
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3831296&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3831296&tab=artifacts#/) on master @ [2b8b0f64c63a376784926667f5fbffa4c4fcc60b](https://github.com/cockroachdb/cockroach/commits/2b8b0f64c63a376784926667f5fbffa4c4fcc60b):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 50899854 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0x4798d80, 0x4b897e, 0xc018954e58, 0xc018954e00)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0xc027917840, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc01f118ea0, {0x61cf9a0, 0xc02514bb00}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:170 +0x6d6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x61cf9a0, 0xc02514bb00}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:90 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0x20300a, 0xc029a8ecc0, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:855 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:804
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x6225a50, 0xc02f1f6720}, {0x62dbdc0, 0xc00ce41b00}, {0xc009312060, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:834 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc016ee3000, {0x6225a50, 0xc02f1f6720}, 0x9)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3ea
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x6225a50, 0xc02f1f6720}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:309 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x6225a50)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:343 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc022fc93b0, {0x6225a50, 0xc01baa18c0}, 0xc0028323c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x6225a50, 0xc01baa18c0}, {0x4917f40, 0xc0028323c0})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x6225a50, 0xc01baa18c0}, {0x4917f40, 0xc0028323c0}, 0xc016730480, 0xc031be67b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6225a50, 0xc01baa18c0}, {0x4917f40, 0xc0028323c0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x6225a50, 0xc01baa18c0}, {0x4917f40, 0xc0028323c0}, 0xc01baa1890, 0xc02606dd00)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6225a50, 0xc01baa18c0}, {0x4917f40, 0xc0028323c0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c016dcbb60}}}, {0x6225a50, 0xc023218810}, {0x4917f40, 0xc0028323c0}, 0xc016730480, 0xc02606dd00)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6225a50, 0xc023218810}, {0x4917f40, 0xc0028323c0})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x6225a50, 0xc023218810}, {0x4917f40, 0xc0028323c0}, 0xc016730480, 0xc031be67b0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x43f7cc0, 0xc022fc93b0}, {0x6225a50, 0xc023218810}, 0xc002832360, 0xc0215de000)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc018f5ba40, {0x62afc20, 0xc00aeebb00}, 0xc02f1a97a0, 0xc032f32ed0, 0x8b1f050, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc018f5ba40, {0x62afc20, 0xc00aeebb00}, 0xc02f1a97a0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs-tenant/multi_region_default_primary_region
--- done: testdata/logic_test/multi_region_default_primary_region with config multiregion-9node-3region-3azs-tenant: 12 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3847976&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3847976&tab=artifacts#/) on master @ [8b696d17241b2eb2d333e23750cf6fdc433897d8](https://github.com/cockroachdb/cockroach/commits/8b696d17241b2eb2d333e23750cf6fdc433897d8):

Fatal error:

```
panic: pebble: closed

```
Stack: 

```
goroutine 52024063 [running]:
github.com/cockroachdb/pebble.(*DB).newIterInternal(0xc038394d80, 0xc038080e58, 0x12e31ec, 0xc0015c0750)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:746 +0x317
github.com/cockroachdb/pebble.(*DB).NewIter(0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/db.go:923 +0x25
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).init(0xc034cbc6c0, {0x61db380, 0xc0440b0000}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:179 +0x7b6
github.com/cockroachdb/cockroach/pkg/storage.newPebbleIterator({0x61db380, 0xc0440b0000}, {0x0, 0x0}, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:91 +0xbb
github.com/cockroachdb/cockroach/pkg/storage.(*Pebble).NewMVCCIterator(0x20300b, 0xc02fa237c0, {0x1, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble.go:965 +0x185
github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:804
github.com/cockroachdb/cockroach/pkg/storage.MVCCGet({0x6231530, 0xc034cb8150}, {0x62e79a0, 0xc043b15600}, {0xc00b9fbf20, 0xa, 0xa}, {0x0, 0x0, 0x0}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:834 +0x123
github.com/cockroachdb/cockroach/pkg/server.(*nodeTombstoneStorage).IsDecommissioned(0xc0440da300, {0x6231530, 0xc034cb8150}, 0x9)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node_tombstone_storage.go:62 +0x3ea
github.com/cockroachdb/cockroach/pkg/server.NewServer.func1({0x6231530, 0xc034cb8150}, 0x1, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:309 +0x71
github.com/cockroachdb/cockroach/pkg/server.NewServer.func3(0x6231530)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:343 +0x31
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc0442dcfa0, {0x6231530, 0xc0160cfa70}, 0xc03a8d7560)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.go:172 +0x498
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x6231530, 0xc0160cfa70}, {0x4920480, 0xc03a8d7560})
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:283 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1({0x6231530, 0xc0160cfa70}, {0x4920480, 0xc03a8d7560}, 0xc01a31ed80, 0xc016c5ca08)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:115 +0x234
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6231530, 0xc0160cfa70}, {0x4920480, 0xc03a8d7560})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1131 +0x63
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1({0x6231530, 0xc0160cfa70}, {0x4920480, 0xc03a8d7560}, 0xc0160cfa40, 0xc02fa23680)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:183 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6231530, 0xc0160cfa70}, {0x4920480, 0xc03a8d7560})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({{{0x10000c01ab1ab60}}}, {0x6231530, 0xc0160cf9e0}, {0x4920480, 0xc03a8d7560}, 0xc01a31ed80, 0xc02fa23680)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:71 +0x173
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x6231530, 0xc0160cf9e0}, {0x4920480, 0xc03a8d7560})
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1134 +0x87
google.golang.org/grpc.chainUnaryInterceptors.func1({0x6231530, 0xc0160cf9e0}, {0x4920480, 0xc03a8d7560}, 0xc01a31ed80, 0xc016c5ca08)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1136 +0x159
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x43ff340, 0xc0442dcfa0}, {0x6231530, 0xc0160cf9e0}, 0xc03a8d7500, 0xc044298920)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:285 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc043ba4e00, {0x62bb700, 0xc03b3adc80}, 0xc01508c120, 0xc039a64810, 0x8b2f070, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1297 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc043ba4e00, {0x62bb700, 0xc03b3adc80}, 0xc01508c120, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1626 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:941 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:939 +0x294
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs-tenant/multi_region_drop_region
--- done: testdata/logic_test/multi_region_drop_region with config multiregion-9node-3region-3azs-tenant: 4 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3908995&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3908995&tab=artifacts#/) on master @ [0c2fea7fc7ed1c61d819681d6c5e153109b3542b](https://github.com/cockroachdb/cockroach/commits/0c2fea7fc7ed1c61d819681d6c5e153109b3542b):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2787503931
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3572: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2787503931
--- FAIL: TestCCLLogic (556.97s)
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value
--- done: testdata/logic_test/partitioning_constrained_scans with config local: 33 tests, 0 failures
--- progress: testdata/logic_test/partitioning: 87 statements/queries
--- progress: testdata/logic_test/partitioning_index: 13 statements/queries
--- progress: testdata/logic_test/partitioning_implicit: 35 statements/queries
--- done: testdata/logic_test/crdb_internal with config local: 17 tests, 0 failures
--- total progress: 152 statements/queries
    logic.go:1979: 
         
        testdata/logic_test/partitioning_enum:92: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: testdata/logic_test/partitioning_enum with config local: 15 tests, 1 failures
--- progress: testdata/logic_test/partitioning: 120 statements/queries
--- done: testdata/logic_test/partitioning_index with config local: 28 tests, 0 failures
--- progress: testdata/logic_test/partitioning_implicit: 47 statements/queries
--- done: testdata/logic_test/partitioning with config local: 138 tests, 0 failures
            --- FAIL: TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value (2.48s)
=== RUN   TestCCLLogic/local/partitioning_enum
=== PAUSE TestCCLLogic/local/partitioning_enum
=== CONT  TestCCLLogic/local/partitioning_enum
        --- FAIL: TestCCLLogic/local/partitioning_enum (6.56s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3986155&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3986155&tab=artifacts#/) on master @ [8d4953ef35bf310c72108c2790ba21971e9e7291](https://github.com/cockroachdb/cockroach/commits/8d4953ef35bf310c72108c2790ba21971e9e7291):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/517f2c78a21274cac19050c7e7164757/logTestCCLLogic422653465
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3593: -- test log scope end --
--- FAIL: TestCCLLogic (321.79s)
=== RUN   TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value
--- done: testdata/logic_test/case_sensitive_names with config local: 2 tests, 0 failures
--- done: testdata/logic_test/changefeed with config local: 7 tests, 0 failures
--- progress: testdata/logic_test/partitioning: 112 statements/queries
--- total progress: 128 statements/queries
--- progress: testdata/logic_test/partitioning_implicit: 32 statements/queries
--- progress: testdata/logic_test/partitioning_index: 19 statements/queries
--- done: testdata/logic_test/crdb_internal with config local: 17 tests, 0 failures
--- done: testdata/logic_test/partitioning_index with config local: 28 tests, 0 failures
--- done: testdata/logic_test/partitioning with config local: 138 tests, 0 failures
    logic.go:1990: 
         
        testdata/logic_test/partitioning_enum:92: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: testdata/logic_test/partitioning_enum with config local: 15 tests, 1 failures
--- total progress: 333 statements/queries
--- progress: testdata/logic_test/partitioning_implicit: 50 statements/queries
            --- FAIL: TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value (2.33s)
=== RUN   TestCCLLogic/local/partitioning_enum
=== PAUSE TestCCLLogic/local/partitioning_enum
=== CONT  TestCCLLogic/local/partitioning_enum
--- done: testdata/logic_test/partitioning_all_by_nothing with config local: 7 tests, 0 failures
--- done: testdata/logic_test/partitioning_constrained_scans with config local: 33 tests, 0 failures
        --- FAIL: TestCCLLogic/local/partitioning_enum (5.01s)
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4048793&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4048793&tab=artifacts#/) on master @ [ced0dd436871aa007c94d5393c26d57aba429196](https://github.com/cockroachdb/cockroach/commits/ced0dd436871aa007c94d5393c26d57aba429196):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic414958105
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3584: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic414958105
--- FAIL: TestCCLLogic (464.61s)
=== RUN   TestCCLLogic/local/partitioning_enum
=== PAUSE TestCCLLogic/local/partitioning_enum
=== CONT  TestCCLLogic/local/partitioning_enum
        --- FAIL: TestCCLLogic/local/partitioning_enum (6.36s)
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value
--- done: testdata/logic_test/partitioning_constrained_scans with config local: 33 tests, 0 failures
--- done: testdata/logic_test/auto_rehoming with config local: 8 tests, 0 failures
--- progress: testdata/logic_test/partitioning: 89 statements/queries
--- progress: testdata/logic_test/partitioning_implicit: 35 statements/queries
--- progress: testdata/logic_test/partitioning_index: 13 statements/queries
--- done: testdata/logic_test/crdb_internal with config local: 17 tests, 0 failures
--- total progress: 160 statements/queries
    logic.go:1981: 
         
        testdata/logic_test/partitioning_enum:92: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: testdata/logic_test/partitioning_enum with config local: 15 tests, 1 failures
--- done: testdata/logic_test/partitioning_index with config local: 28 tests, 0 failures
--- progress: testdata/logic_test/partitioning: 129 statements/queries
--- progress: testdata/logic_test/partitioning_implicit: 47 statements/queries
--- done: testdata/logic_test/partitioning with config local: 138 tests, 0 failures
            --- FAIL: TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value (2.27s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4140502&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4140502&tab=artifacts#/) on master @ [365b4da8bd02c06ee59d2130a56dec74ffc9ce21](https://github.com/cockroachdb/cockroach/commits/365b4da8bd02c06ee59d2130a56dec74ffc9ce21):

Fatal error:

```
panic: getting first from empty ring buffer

```
Stack: 

```
goroutine 432535 [running]:
github.com/cockroachdb/cockroach/pkg/util/ring.(*Buffer).GetFirst(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ring/ring_buffer.go:69
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordInternalLocked(0xc01d1d2c50, {0x5ef9c20, 0xc02b434280}, 0xc01d1d2d28)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:478 +0x205
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordFinishedChildrenLocked(0xc01d1d2c50, {0xc025585e00, 0x1, 0x44e5ad})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:380 +0x29e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordFinishedChildren(0xbb2bf5, {0xc025585e00, 0x4578e80, 0x8e0d378})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:360 +0xbe
github.com/cockroachdb/cockroach/pkg/util/tracing.(*spanInner).ImportRemoteSpans(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/span_inner.go:80
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).ImportRemoteSpans(0xc01d1d2c00, {0xc025585e00, 0x1, 0x3})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:180 +0x4e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).pushMeta(0xc0204db500, 0xc023273110)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:967 +0x33e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).Push(0xc0204db500, {0x0, 0x0, 0x0}, 0x2dbcfa0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1047 +0x29c
github.com/cockroachdb/cockroach/pkg/sql/execinfra.Run({0x5f96380, 0xc00fae4400}, {0x5fbf5f8, 0xc029faee00}, {0x5f195b8, 0xc0204db500})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/base.go:189 +0x86
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc029faee00, {0x5f96380, 0xc00fae4400})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:733 +0x5b
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc00709c4e0, {0x5f96380, 0xc00fae4400}, 0xc029faee00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:453 +0x258
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc0182da660, {0x5f96380, 0xc00fae4400}, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:246 +0x205
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc02aea0500, 0xc006c34a50, 0xc00b981760, 0xc00caa9680, 0xc0204db500, 0xc01d1d2f00, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:563 +0xb56
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).planAndRunSubquery(0xc02aea0500, {0x5fb2460, 0xc01d1d2da0}, 0x0, {{0x5ef90e0, 0xc0163f4180}, 0x2, 0x1, 0x0, {{0x5f989c8, ...}, ...}, ...}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1269 +0x7b7
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunSubqueries(0x5fb2460, {0x5fb2460, 0xc01d1d2da0}, 0x1, 0xc02dde67c0, {0xc00ce4e9b0, 0x3, 0x1}, 0xc00b981760, 0xc02dde6788)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1174 +0x17b
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc017576c00, {0x5fb2460, 0xc01d1d2da0}, 0xc017577248, 0xc01d1d2da0, {0x6037b68, 0xc019100e10}, 0x0, 0xc0022a95b8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1398 +0x566
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc017576c00, {0x5fb2460, 0xc01d1d2da0}, 0xc017577248, {0x6037b68, 0xc019100e10})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1084 +0x9c7
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc017576c00, {0x5fb2460, 0xc01d1d2da0}, {{0x5fc61b8, 0xc017640500}, {0xc01cd26000, 0xde26}, 0x12e, 0x12e}, 0xc01b70fa00, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:721 +0x2526
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc017576c00, {0x5fb2460, 0xc01d1d27a0}, {{0x5fc61b8, 0xc017640500}, {0xc01cd26000, 0xde26}, 0x12e, 0x12e}, 0xc01b70fa00, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:135 +0x572
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execPortal(0xc017576c00, {0x5fb2460, 0xc01d1d27a0}, {0xc01b70fa00, {0xc007558600, 0x12e, 0x12e}, {0x0, 0x0, 0x0}, ...}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:222 +0x158
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func2({{0x0, 0x0}, 0x0, {0x1582cc08, 0xed97853ab, 0x0}}, 0xc017576c00, 0xc02dde7bb8, 0xc02dde7ba8, 0xc02539fac8, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1820 +0x511
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc017576c00, {0x5f96380, 0xc0173e94c0})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1822 +0x43a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc017576c00, {0x5fb2460, 0xc01ccdf3a0}, 0xc01031a7d0, {0xae2318, 0xc0177e2380, 0x0}, 0xc01dcdeea0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1676 +0x285
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:205 +0x8d
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:204 +0x5f1
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_query_behavior/regressions
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4149825&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4149825&tab=artifacts#/) on master @ [912964e02ddd951c77d4f71981ae18b3894e9084](https://github.com/cockroachdb/cockroach/commits/912964e02ddd951c77d4f71981ae18b3894e9084):

Fatal error:

```
panic: getting first from empty ring buffer

```
Stack: 

```
goroutine 432973 [running]:
github.com/cockroachdb/cockroach/pkg/util/ring.(*Buffer).GetFirst(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ring/ring_buffer.go:69
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordInternalLocked(0xc02a594350, {0x5f077a0, 0xc013fe8fc0}, 0xc02a594428)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:478 +0x205
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordFinishedChildrenLocked(0xc02a594350, {0xc02e5461e0, 0x1, 0x44e80d})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:380 +0x29e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordFinishedChildren(0xbb30b5, {0xc02e5461e0, 0x45811c0, 0x8e20978})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:360 +0xbe
github.com/cockroachdb/cockroach/pkg/util/tracing.(*spanInner).ImportRemoteSpans(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/span_inner.go:80
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).ImportRemoteSpans(0xc02a594300, {0xc02e5461e0, 0x1, 0x3})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:180 +0x4e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).pushMeta(0xc033a1a380, 0xc01465f6c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:967 +0x33e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).Push(0xc033a1a380, {0x0, 0x0, 0x0}, 0x2dc2200)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1047 +0x29c
github.com/cockroachdb/cockroach/pkg/sql/execinfra.Run({0x5fa41c0, 0xc00e19fe00}, {0x5fcd5d8, 0xc028ca0d00}, {0x5f27138, 0xc033a1a380})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/base.go:189 +0x86
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc028ca0d00, {0x5fa41c0, 0xc00e19fe00})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:733 +0x5b
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc015581520, {0x5fa41c0, 0xc00e19fe00}, 0xc028ca0d00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:453 +0x258
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc00128ab70, {0x5fa41c0, 0xc00e19fe00}, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:246 +0x205
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc0109c8000, 0xc007c1de00, 0xc00fc556b0, 0xc01b6bc300, 0xc033a1a380, 0xc02a594600, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:563 +0xb56
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).planAndRunSubquery(0xc0109c8000, {0x5fc0498, 0xc02a5944a0}, 0x0, {{0x5f06c60, 0xc0227dc9f0}, 0x2, 0x1, 0x0, {{0x5fa68b0, ...}, ...}, ...}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1269 +0x7b7
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunSubqueries(0x5fc0498, {0x5fc0498, 0xc02a5944a0}, 0x1, 0xc035b647c0, {0xc00e579130, 0x3, 0x1}, 0xc00fc556b0, 0xc035b64788)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1174 +0x17b
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc01e291300, {0x5fc0498, 0xc02a5944a0}, 0xc01e291948, 0xc02a5944a0, {0x6045bc8, 0xc01d679ef0}, 0x0, 0xc018016dd8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1398 +0x566
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc01e291300, {0x5fc0498, 0xc02a5944a0}, 0xc01e291948, {0x6045bc8, 0xc01d679ef0})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1084 +0x9c7
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc01e291300, {0x5fc0498, 0xc02a5944a0}, {{0x5fd4198, 0xc0133c6690}, {0xc01b878000, 0xde26}, 0x12e, 0x12e}, 0xc00eacd100, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:721 +0x2526
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc01e291300, {0x5fc0498, 0xc02a5941a0}, {{0x5fd4198, 0xc0133c6690}, {0xc01b878000, 0xde26}, 0x12e, 0x12e}, 0xc00eacd100, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:135 +0x572
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execPortal(0xc01e291300, {0x5fc0498, 0xc02a5941a0}, {0xc00eacd100, {0xc014164600, 0x12e, 0x12e}, {0x0, 0x0, 0x0}, ...}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:222 +0x158
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func2({{0x0, 0x0}, 0x0, {0x1bb634e3, 0xed979a560, 0x0}}, 0xc01e291300, 0xc035b65bb8, 0xc035b65ba8, 0xc01dddfac8, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1820 +0x511
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc01e291300, {0x5fa41c0, 0xc03241e240})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1822 +0x43a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc01e291300, {0x5fc0498, 0xc0389770a0}, 0xc0146cefd0, {0xae27d8, 0xc022b73880, 0x0}, 0xc03068d380)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1676 +0x285
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:205 +0x8d
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:204 +0x5f1
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/regional_by_row_query_behavior/regressions
--- done: testdata/logic_test/regional_by_row_query_behavior with config multiregion-9node-3region-3azs: 144 tests, 0 failures
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4160116&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4160116&tab=artifacts#/) on master @ [9b0065ca2450205a34d4237be7b317c2d895658c](https://github.com/cockroachdb/cockroach/commits/9b0065ca2450205a34d4237be7b317c2d895658c):

Fatal error:

```
panic: getting first from empty ring buffer

```
Stack: 

```
goroutine 279373 [running]:
github.com/cockroachdb/cockroach/pkg/util/ring.(*Buffer).GetFirst(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ring/ring_buffer.go:69
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordInternalLocked(0xc007348050, {0x5f18100, 0xc00a35fcc0}, 0xc007348128)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:478 +0x205
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordFinishedChildrenLocked(0xc007348050, {0xc01cf0fa40, 0x1, 0x44e80d})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:380 +0x29e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*crdbSpan).recordFinishedChildren(0xbb30b5, {0xc01cf0fa40, 0x45886e0, 0x8e39e78})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/crdbspan.go:360 +0xbe
github.com/cockroachdb/cockroach/pkg/util/tracing.(*spanInner).ImportRemoteSpans(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/span_inner.go:80
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).ImportRemoteSpans(0xc007348000, {0xc01cf0fa40, 0x1, 0x3})
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:180 +0x4e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).pushMeta(0xc018caa700, 0xc017f523f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:967 +0x33e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).Push(0xc018caa700, {0x0, 0x0, 0x0}, 0x2dc7760)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1047 +0x29c
github.com/cockroachdb/cockroach/pkg/sql/execinfra.Run({0x5fb4ac0, 0xc00cc04740}, {0x5fdded8, 0xc00a309d00}, {0x5f37a98, 0xc018caa700})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/base.go:189 +0x86
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc00a309d00, {0x5fb4ac0, 0xc00cc04740})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:733 +0x5b
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc0189eab60, {0x5fb4ac0, 0xc00cc04740}, 0xc00a309d00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:453 +0x258
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc00d02a750, {0x5fb4ac0, 0xc00cc04740}, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:246 +0x205
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc01d6d9600, 0xc006e105a0, 0xc01d1eaa50, 0xc013826a00, 0xc018caa700, 0xc007348300, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:563 +0xb56
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).planAndRunSubquery(0xc01d6d9600, {0x5fd0d98, 0xc0073481a0}, 0x0, {{0x5f175c0, 0xc00c16f650}, 0x2, 0x1, 0x0, {{0x5fb71b0, ...}, ...}, ...}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1269 +0x7b7
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunSubqueries(0x5fd0d98, {0x5fd0d98, 0xc0073481a0}, 0x1, 0xc01fb407c0, {0xc00cf5c730, 0x3, 0x1}, 0xc01d1eaa50, 0xc01fb40788)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1174 +0x17b
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc017502000, {0x5fd0d98, 0xc0073481a0}, 0xc017502648, 0xc0073481a0, {0x6056648, 0xc009e17b90}, 0x0, 0xc02d926e98)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1398 +0x566
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc017502000, {0x5fd0d98, 0xc0073481a0}, 0xc017502648, {0x6056648, 0xc009e17b90})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1084 +0x9c7
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc017502000, {0x5fd0d98, 0xc0073481a0}, {{0x5fe4b18, 0xc01dd5aa00}, {0xc007ed2000, 0xa316}, 0xde, 0xde}, 0xc0112b0200, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:721 +0x2526
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc017502000, {0x5fd0d98, 0xc00c465ca0}, {{0x5fe4b18, 0xc01dd5aa00}, {0xc007ed2000, 0xa316}, 0xde, 0xde}, 0xc0112b0200, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:135 +0x572
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execPortal(0xc017502000, {0x5fd0d98, 0xc00c465ca0}, {0xc0112b0200, {0xc007332000, 0xde, 0xde}, {0x0, 0x0, 0x0}, ...}, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:222 +0x158
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func2({{0x0, 0x0}, 0x0, {0x2ad9e81f, 0xed97af62c, 0x0}}, 0xc017502000, 0xc01fb41bb8, 0xc01fb41ba8, 0xc01b121ac8, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1820 +0x511
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc017502000, {0x5fb4ac0, 0xc0064d6bc0})
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1822 +0x43a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc017502000, {0x5fd0d98, 0xc00bc51ca0}, 0xc02333f300, {0x5f1b780, 0xc011510f80, 0x0}, 0xc01727e840)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1676 +0x285
github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:205 +0x8d
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).initConnEx
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:204 +0x5f1
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/multi_region_backup/restore_tables_into_database_with_different_regions
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4171309&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4171309&tab=artifacts#/) on master @ [488a06bb5c480e7ea47bbe86f006912850996b7b](https://github.com/cockroachdb/cockroach/commits/488a06bb5c480e7ea47bbe86f006912850996b7b):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2818673249
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3604: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2818673249
--- FAIL: TestCCLLogic (570.47s)
=== RUN   TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_constrained_scans with config local: 33 tests, 0 failures
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_implicit: 17 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/auto_rehoming: 7 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/crdb_internal: 7 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/auto_rehoming with config local: 8 tests, 0 failures
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning: 95 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_index: 13 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/crdb_internal with config local: 17 tests, 0 failures
--- total progress: 143 statements/queries
    logic.go:1985: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config local: 14 tests, 1 failures
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_implicit: 44 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning: 120 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_index: 24 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_index with config local: 28 tests, 0 failures
--- total progress: 202 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_implicit: 54 statements/queries
            --- FAIL: TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value (2.33s)
=== RUN   TestCCLLogic/local/partitioning_enum
=== PAUSE TestCCLLogic/local/partitioning_enum
=== CONT  TestCCLLogic/local/partitioning_enum
        --- FAIL: TestCCLLogic/local/partitioning_enum (8.73s)
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4172533&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4172533&tab=artifacts#/) on master @ [58ceac139a7e83052171121b28026a7366f16f7e](https://github.com/cockroachdb/cockroach/commits/58ceac139a7e83052171121b28026a7366f16f7e):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic764452161
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3604: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic764452161
--- FAIL: TestCCLLogic (250.73s)
=== RUN   TestCCLLogic/fakedist
    --- FAIL: TestCCLLogic/fakedist (0.00s)
=== RUN   TestCCLLogic/fakedist/partitioning_enum/drop_enum_partitioning_value
--- total progress: 475 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_constrained_scans with config fakedist: 33 tests, 0 failures
    logic.go:1985: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist: 14 tests, 1 failures
            --- FAIL: TestCCLLogic/fakedist/partitioning_enum/drop_enum_partitioning_value (1.12s)
=== RUN   TestCCLLogic/fakedist/partitioning_enum
=== PAUSE TestCCLLogic/fakedist/partitioning_enum
=== CONT  TestCCLLogic/fakedist/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist/partitioning_enum (3.23s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4172298&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4172298&tab=artifacts#/) on master @ [58ceac139a7e83052171121b28026a7366f16f7e](https://github.com/cockroachdb/cockroach/commits/58ceac139a7e83052171121b28026a7366f16f7e):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2520609772
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3604: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2520609772
--- FAIL: TestCCLLogic (516.81s)
=== RUN   TestCCLLogic/fakedist-vec-off/partitioning_enum/drop_enum_partitioning_value
    logic.go:1985: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist-vec-off: 14 tests, 1 failures
--- total progress: 551 statements/queries
            --- FAIL: TestCCLLogic/fakedist-vec-off/partitioning_enum/drop_enum_partitioning_value (1.62s)
=== RUN   TestCCLLogic/fakedist-vec-off/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-vec-off/partitioning_enum
=== CONT  TestCCLLogic/fakedist-vec-off/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-vec-off/partitioning_enum (4.82s)
=== RUN   TestCCLLogic/fakedist-vec-off
    --- FAIL: TestCCLLogic/fakedist-vec-off (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4175456&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4175456&tab=artifacts#/) on master @ [72655fd4730fadb0ddacd8a5513a5d2d9121cbc6](https://github.com/cockroachdb/cockroach/commits/72655fd4730fadb0ddacd8a5513a5d2d9121cbc6):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic1797527682
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic1797527682
--- FAIL: TestCCLLogic (512.15s)
=== RUN   TestCCLLogic/local-vec-off/partitioning_enum/drop_enum_partitioning_value
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config local-vec-off: 14 tests, 1 failures
--- total progress: 446 statements/queries
            --- FAIL: TestCCLLogic/local-vec-off/partitioning_enum/drop_enum_partitioning_value (1.31s)
=== RUN   TestCCLLogic/local-vec-off/partitioning_enum
=== PAUSE TestCCLLogic/local-vec-off/partitioning_enum
=== CONT  TestCCLLogic/local-vec-off/partitioning_enum
        --- FAIL: TestCCLLogic/local-vec-off/partitioning_enum (3.45s)
=== RUN   TestCCLLogic/local-vec-off
    --- FAIL: TestCCLLogic/local-vec-off (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4177652&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4177652&tab=artifacts#/) on master @ [6264a82d276489bc11ad87bc98fd237319501c9f](https://github.com/cockroachdb/cockroach/commits/6264a82d276489bc11ad87bc98fd237319501c9f):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2680317562
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2680317562
--- FAIL: TestCCLLogic (516.67s)
=== RUN   TestCCLLogic/fakedist/partitioning_enum/drop_enum_partitioning_value
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_constrained_scans with config fakedist: 33 tests, 0 failures
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist: 14 tests, 1 failures
--- total progress: 527 statements/queries
            --- FAIL: TestCCLLogic/fakedist/partitioning_enum/drop_enum_partitioning_value (1.60s)
=== RUN   TestCCLLogic/fakedist/partitioning_enum
=== PAUSE TestCCLLogic/fakedist/partitioning_enum
=== CONT  TestCCLLogic/fakedist/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist/partitioning_enum (5.00s)
=== RUN   TestCCLLogic/fakedist
    --- FAIL: TestCCLLogic/fakedist (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4178384&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4178384&tab=artifacts#/) on master @ [6264a82d276489bc11ad87bc98fd237319501c9f](https://github.com/cockroachdb/cockroach/commits/6264a82d276489bc11ad87bc98fd237319501c9f):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic3920000137
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic3920000137
--- FAIL: TestCCLLogic (555.03s)
=== RUN   TestCCLLogic/fakedist/partitioning_enum/drop_enum_partitioning_value
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist: 14 tests, 1 failures
--- total progress: 527 statements/queries
            --- FAIL: TestCCLLogic/fakedist/partitioning_enum/drop_enum_partitioning_value (1.47s)
=== RUN   TestCCLLogic/fakedist/partitioning_enum
=== PAUSE TestCCLLogic/fakedist/partitioning_enum
=== CONT  TestCCLLogic/fakedist/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist/partitioning_enum (6.12s)
=== RUN   TestCCLLogic/fakedist
    --- FAIL: TestCCLLogic/fakedist (0.00s)
=== RUN   TestCCLLogic/fakedist-vec-off/partitioning_enum/drop_enum_partitioning_value
--- total progress: 532 statements/queries
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist-vec-off: 14 tests, 1 failures
--- total progress: 549 statements/queries
            --- FAIL: TestCCLLogic/fakedist-vec-off/partitioning_enum/drop_enum_partitioning_value (1.74s)
=== RUN   TestCCLLogic/fakedist-vec-off/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-vec-off/partitioning_enum
=== CONT  TestCCLLogic/fakedist-vec-off/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-vec-off/partitioning_enum (5.12s)
=== RUN   TestCCLLogic/fakedist-vec-off
    --- FAIL: TestCCLLogic/fakedist-vec-off (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4179205&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4179205&tab=artifacts#/) on master @ [22f45e597c8f698a556f2ae32ff68e4c0327e97a](https://github.com/cockroachdb/cockroach/commits/22f45e597c8f698a556f2ae32ff68e4c0327e97a):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic67507952
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic67507952
--- FAIL: TestCCLLogic (515.56s)
=== RUN   TestCCLLogic/local-vec-off/partitioning_enum/drop_enum_partitioning_value
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/new_schema_changer with config local-vec-off: 3 tests, 0 failures
--- total progress: 432 statements/queries
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config local-vec-off: 14 tests, 1 failures
            --- FAIL: TestCCLLogic/local-vec-off/partitioning_enum/drop_enum_partitioning_value (1.61s)
=== RUN   TestCCLLogic/local-vec-off/partitioning_enum
=== PAUSE TestCCLLogic/local-vec-off/partitioning_enum
=== CONT  TestCCLLogic/local-vec-off/partitioning_enum
        --- FAIL: TestCCLLogic/local-vec-off/partitioning_enum (3.91s)
=== RUN   TestCCLLogic/local-vec-off
    --- FAIL: TestCCLLogic/local-vec-off (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4180723&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4180723&tab=artifacts#/) on master @ [1f819bb12af8e9d736f22bb8135f29969732071d](https://github.com/cockroachdb/cockroach/commits/1f819bb12af8e9d736f22bb8135f29969732071d):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic4028218298
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic4028218298
--- FAIL: TestCCLLogic (485.52s)
=== RUN   TestCCLLogic/fakedist-vec-off/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-vec-off/partitioning_enum
=== CONT  TestCCLLogic/fakedist-vec-off/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-vec-off/partitioning_enum (4.76s)
=== RUN   TestCCLLogic/fakedist-vec-off
    --- FAIL: TestCCLLogic/fakedist-vec-off (0.01s)
=== RUN   TestCCLLogic/fakedist-vec-off/partitioning_enum/drop_enum_partitioning_value
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist-vec-off: 14 tests, 1 failures
--- total progress: 551 statements/queries
            --- FAIL: TestCCLLogic/fakedist-vec-off/partitioning_enum/drop_enum_partitioning_value (1.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4181173&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4181173&tab=artifacts#/) on master @ [b88739e06ed3882a55f1a46a2e77dd9463b2f734](https://github.com/cockroachdb/cockroach/commits/b88739e06ed3882a55f1a46a2e77dd9463b2f734):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic4067297948
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic4067297948
--- FAIL: TestCCLLogic (480.11s)
=== RUN   TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_constrained_scans with config local: 33 tests, 0 failures
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/auto_rehoming with config local: 8 tests, 0 failures
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/crdb_internal with config local: 17 tests, 0 failures
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_implicit: 29 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning: 104 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_index: 14 statements/queries
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config local: 14 tests, 1 failures
--- total progress: 174 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_index with config local: 28 tests, 0 failures
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning: 130 statements/queries
--- progress: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_implicit: 47 statements/queries
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning with config local: 138 tests, 0 failures
            --- FAIL: TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value (1.83s)
=== RUN   TestCCLLogic/local/partitioning_enum
=== PAUSE TestCCLLogic/local/partitioning_enum
=== CONT  TestCCLLogic/local/partitioning_enum
        --- FAIL: TestCCLLogic/local/partitioning_enum (6.85s)
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/fakedist-disk/partitioning_enum/drop_enum_partitioning_value
--- total progress: 580 statements/queries
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist-disk: 14 tests, 1 failures
            --- FAIL: TestCCLLogic/fakedist-disk/partitioning_enum/drop_enum_partitioning_value (1.27s)
=== RUN   TestCCLLogic/fakedist-disk/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-disk/partitioning_enum
=== CONT  TestCCLLogic/fakedist-disk/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-disk/partitioning_enum (4.26s)
=== RUN   TestCCLLogic/fakedist-disk
    --- FAIL: TestCCLLogic/fakedist-disk (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4181674&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4181674&tab=artifacts#/) on master @ [dc07599dc9db1acd5afa3a6537297815f25c1fca](https://github.com/cockroachdb/cockroach/commits/dc07599dc9db1acd5afa3a6537297815f25c1fca):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic1365801639
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic1365801639
--- FAIL: TestCCLLogic (246.22s)
=== RUN   TestCCLLogic/fakedist-disk/partitioning_enum/drop_enum_partitioning_value
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist-disk: 14 tests, 1 failures
            --- FAIL: TestCCLLogic/fakedist-disk/partitioning_enum/drop_enum_partitioning_value (0.55s)
=== RUN   TestCCLLogic/fakedist-disk/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-disk/partitioning_enum
=== CONT  TestCCLLogic/fakedist-disk/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-disk/partitioning_enum (1.99s)
=== RUN   TestCCLLogic/fakedist-disk
    --- FAIL: TestCCLLogic/fakedist-disk (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4185046&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4185046&tab=artifacts#/) on master @ [6aa6c727b1d990bc0e3e8fbc36e25fc358ba39c1](https://github.com/cockroachdb/cockroach/commits/6aa6c727b1d990bc0e3e8fbc36e25fc358ba39c1):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2804157900
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic2804157900
--- FAIL: TestCCLLogic (513.16s)
=== RUN   TestCCLLogic/fakedist-disk/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-disk/partitioning_enum
=== CONT  TestCCLLogic/fakedist-disk/partitioning_enum
        --- FAIL: TestCCLLogic/fakedist-disk/partitioning_enum (5.46s)
=== RUN   TestCCLLogic/fakedist-disk
    --- FAIL: TestCCLLogic/fakedist-disk (0.01s)
=== RUN   TestCCLLogic/fakedist-disk/partitioning_enum/drop_enum_partitioning_value
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist-disk: 14 tests, 1 failures
--- total progress: 599 statements/queries
            --- FAIL: TestCCLLogic/fakedist-disk/partitioning_enum/drop_enum_partitioning_value (1.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4185086&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4185086&tab=artifacts#/) on master @ [e1068d77afbd39b162978281c9da7cbea49c1c3a](https://github.com/cockroachdb/cockroach/commits/e1068d77afbd39b162978281c9da7cbea49c1c3a):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic3894224798
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic3894224798
--- FAIL: TestCCLLogic (508.27s)
=== RUN   TestCCLLogic/fakedist-spec-planning/partitioning_enum/drop_enum_partitioning_value
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config fakedist-spec-planning: 14 tests, 1 failures
--- total progress: 623 statements/queries
            --- FAIL: TestCCLLogic/fakedist-spec-planning/partitioning_enum/drop_enum_partitioning_value (1.41s)
=== RUN   TestCCLLogic/fakedist-spec-planning/partitioning_enum
=== PAUSE TestCCLLogic/fakedist-spec-planning/partitioning_enum
=== CONT  TestCCLLogic/fakedist-spec-planning/partitioning_enum
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/tenant_usage with config fakedist-spec-planning: 5 tests, 0 failures
        --- FAIL: TestCCLLogic/fakedist-spec-planning/partitioning_enum (5.09s)
=== RUN   TestCCLLogic/fakedist-spec-planning
    --- FAIL: TestCCLLogic/fakedist-spec-planning (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/logictestccl.TestCCLLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4186392&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4186392&tab=artifacts#/) on master @ [e1068d77afbd39b162978281c9da7cbea49c1c3a](https://github.com/cockroachdb/cockroach/commits/e1068d77afbd39b162978281c9da7cbea49c1c3a):


```
=== RUN   TestCCLLogic
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic3415432435
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic3415432435
--- FAIL: TestCCLLogic (242.27s)
=== RUN   TestCCLLogic/local/partitioning_enum
=== PAUSE TestCCLLogic/local/partitioning_enum
=== CONT  TestCCLLogic/local/partitioning_enum
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_index with config local: 28 tests, 0 failures
        --- FAIL: TestCCLLogic/local/partitioning_enum (3.54s)
=== RUN   TestCCLLogic/local
    --- FAIL: TestCCLLogic/local (0.00s)
=== RUN   TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value
=== CONT  TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value
    logic.go:1994: 
         
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum:93: 
        expected "could not remove enum value \"a\" as it is being used in the partitioning of index tbl@idx", but no error occurred
--- done: /go/src/github.com/cockroachdb/cockroach/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum with config local: 14 tests, 1 failures
            --- FAIL: TestCCLLogic/local/partitioning_enum/drop_enum_partitioning_value (1.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum
+++ b/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum
@@ -86,13 +86,5 @@ create table tbl (i INT, k t,  PRIMARY KEY (i, k), INDEX idx (k) PARTITION BY RA
 statement error pgcode 2BP01 could not remove enum value "a" as it is being used in the partitioning of index tbl@idx
 alter type t drop value 'a';
 
-# We want to fail even if the index is being dropped. The reason is that it
-# would be bad if we rolled back the dropping of the index.
-statement error pgcode XXA00 could not remove enum value "a" as it is being used in the partitioning of index tbl@idx
-begin;
-drop index tbl@idx;
-alter type t drop value 'a';
-commit
-
 statement ok
 alter type t drop value 'c';

--- a/pkg/ccl/partitionccl/BUILD.bazel
+++ b/pkg/ccl/partitionccl/BUILD.bazel
@@ -79,8 +79,10 @@ go_test(
         "//pkg/util/protoutil",
         "//pkg/util/randutil",
         "//pkg/util/uuid",
+        "@com_github_cockroachdb_cockroach_go_v2//crdb",
         "@com_github_cockroachdb_errors//:errors",
         "@com_github_gogo_protobuf//proto",
+        "@com_github_stretchr_testify//require",
         "@in_gopkg_yaml_v2//:yaml_v2",
     ],
 )

--- a/pkg/ccl/partitionccl/partition_test.go
+++ b/pkg/ccl/partitionccl/partition_test.go
@@ -18,11 +18,14 @@ import (
 	"testing"
 	"time"
 
+	"github.com/cockroachdb/cockroach-go/v2/crdb"
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/ccl/importccl"
 	"github.com/cockroachdb/cockroach/pkg/ccl/utilccl"
 	"github.com/cockroachdb/cockroach/pkg/config"
 	"github.com/cockroachdb/cockroach/pkg/config/zonepb"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
@@ -49,6 +52,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/randutil"
 	"github.com/cockroachdb/errors"
 	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/require"
 	yaml "gopkg.in/yaml.v2"
 )
 
@@ -1536,4 +1540,101 @@ func TestRemovePartitioningExpiredLicense(t *testing.T) {
 	expectErr(`ALTER INDEX t@i PARTITION BY RANGE (a) (PARTITION p45 VALUES FROM (4) TO (5))`, partitionErr)
 	expectErr(`ALTER INDEX t@t_pkey CONFIGURE ZONE USING DEFAULT`, zoneErr)
 	expectErr(`ALTER INDEX t@i CONFIGURE ZONE USING DEFAULT`, zoneErr)
+}
+
+// Test that dropping an enum value fails if there's a concurrent index drop
+// for an index partitioned by that enum value. The reason is that it
+// would be bad if we rolled back the dropping of the index.
+func TestDropEnumValueWithConcurrentPartitionedIndexDrop(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	var s serverutils.TestServerInterface
+	var sqlDB *gosql.DB
+
+	// Use the dropCh to block any DROP INDEX job until the channel is closed.
+	dropCh := make(chan chan struct{})
+	ctx, cancel := context.WithCancel(context.Background())
+	s, sqlDB, _ = serverutils.StartServer(t, base.TestServerArgs{
+		Knobs: base.TestingKnobs{
+			SQLSchemaChanger: &sql.SchemaChangerTestingKnobs{
+				RunBeforeResume: func(jobID jobspb.JobID) error {
+					var isDropJob bool
+					if err := sqlDB.QueryRow(`
+SELECT count(*) > 0
+  FROM [SHOW JOB $1]
+ WHERE description LIKE 'DROP INDEX %'
+`, jobID).Scan(&isDropJob); err != nil {
+						return err
+					}
+					if !isDropJob {
+						return nil
+					}
+					ch := make(chan struct{})
+					select {
+					case dropCh <- ch:
+					case <-ctx.Done():
+						return ctx.Err()
+					}
+					select {
+					case <-ch:
+					case <-ctx.Done():
+						return ctx.Err()
+					}
+					return nil
+				},
+			},
+		},
+	})
+	defer s.Stopper().Stop(context.Background())
+	defer cancel()
+	tdb := sqlutils.MakeSQLRunner(sqlDB)
+
+	// Set up the table to have an index which is partitioned by the enum value
+	// we're going to drop.
+	for _, stmt := range []string{
+		`CREATE TYPE t AS ENUM ('a', 'b', 'c')`,
+		`CREATE TABLE tbl (
+    i INT8, k t,
+    PRIMARY KEY (i, k),
+    INDEX idx (k)
+        PARTITION BY RANGE (k)
+            (PARTITION a VALUES FROM ('a') TO ('b'))
+)`,
+	} {
+		tdb.Exec(t, stmt)
+	}
+	// Run a transaction to drop the index and the enum value.
+	errCh := make(chan error)
+	go func() {
+		errCh <- crdb.ExecuteTx(ctx, sqlDB, nil, func(tx *gosql.Tx) error {
+			if _, err := tx.Exec("drop index tbl@idx;"); err != nil {
+				return err
+			}
+			_, err := tx.Exec("alter type t drop value 'a';")
+			return err
+		})
+	}()
+	// Wait until the dropping of the enum value has finished.
+	ch := <-dropCh
+	testutils.SucceedsSoon(t, func() error {
+		var done bool
+		tdb.QueryRow(t, `
+SELECT bool_and(done)
+  FROM (
+        SELECT status NOT IN `+jobs.NonTerminalStatusTupleString+` AS done
+          FROM [SHOW JOBS]
+         WHERE job_type = 'TYPEDESC SCHEMA CHANGE'
+       );`).
+			Scan(&done)
+		if done {
+			return nil
+		}
+		return errors.Errorf("not done")
+	})
+	// Allow the dropping of the index to proceed.
+	close(ch)
+	// Ensure we got the right error.
+	require.Regexp(t,
+		`could not remove enum value "a" as it is being used in the partitioning of index tbl@idx`,
+		<-errCh)
 }

